### PR TITLE
[OptionsResolver] Remove deprecated code

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by Nyholm — APPROVED** [review]
Thank you

**Comment by derrabus** [comment]
Can you please rebase to see if the CI turns green then?

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php [hunk 1]
```diff
@@ -461,7 +461,7 @@ public function testFailIfSetDeprecatedFromLazyOption()
     public function testSetDeprecatedFailsIfUnknownOption()
     {
         $this->expectException(UndefinedOptionsException::class);
-        $this->resolver->setDeprecated('foo');
+        $this->resolver->setDeprecated('foo', 'vendor/package', '1.1');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php:464]: these could be worth fixing in 5.2,PR welcome

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request merged**